pkg/sql/execinfra/utils.go

```diff
@@ -32,3 +36,15 @@ func DecodeDatum(datumAlloc *tree.DatumAlloc, typ *types.T, data []byte) (tree.D
 	}
 	return datum, nil
 }
+
+// HydrateTypesInDatumInfo hydrates all user-defined types in the provided
+// DatumInfo slice.
+func HydrateTypesInDatumInfo(ctx context.Context, f *FlowCtx, info []execinfrapb.DatumInfo) error {
+	resolver := f.NewTypeResolver(f.Txn)
+	for i := range info {
```

Comment:
nit: lazily construct this thing so you don't have to construct it if there are no UDTs? Also, it feels like this thing should just be a field in the `FlowCtx` given it just has two members which are both members of the `FlowCtx`, no?